### ISSUE TITLE
Building PM（Prediction Module） of DSSD with gluoncv

### ISSUE THREAD

**Issue by Lxh1992** [dmlc/gluon-cv#783]
When I build the Prediction Module of DSSD with gluoncv, the training loss become NAN and I don't know the reason. I will be very gratitude if anyone could help me with this problem. The code is as follows:

`class NewConvPredictor(HybridBlock):
    """Convolutional predictor.
    Convolutional predictor is widely used in object-detection. It can be used
    to predict classification scores (1 channel per class) or box predictor,
    which is usually 4 channels per box.
    The output is of shape (N, num_channel, H, W).

    Parameters
    ----------
    num_channel : int
        Number of conv channels.
    kernel : tuple of (int, int), default (3, 3)
        Conv kernel size as (H, W).
    pad : tuple of (int, int), default (1, 1)
        Conv padding size as (H, W).
    stride : tuple of (int, int), default (1, 1)
        Conv stride size as (H, W).
    activation : str, optional
        Optional activation after conv, e.g. 'relu'.
    use_bias : bool
        Use bias in convolution. It is not necessary if BatchNorm is followed.
    in_channels : int, default is 0
        The number of input channels to this layer.
        If not specified, initialization will be deferred to the first time forward is called and
        in_channels will be inferred from the shape of input data.

    """
    def __init__(self, num_channel, kernel=(3, 3), pad=(1, 1), stride=(1, 1),
                 activation=None, use_bias=True, in_channels=0, **kwargs):
        super(NewConvPredictor, self).__init__(**kwargs)
        with self.name_scope():
            self.conv1 = nn.Conv2D(
                1024, (1,1), strides=stride, padding=(0,0),
                activation=activation, use_bias=use_bias, in_channels=in_channels,
                weight_initializer=mx.init.Xavier(magnitude=2),
                bias_initializer='zeros')
            self.conv2_1 = nn.Conv2D(
                256, (1,1), strides=stride, padding=(0,0),
                activation=activation, use_bias=use_bias, in_channels=in_channels,
                weight_initializer=mx.init.Xavier(magnitude=2),
                bias_initializer='zeros')
            self.conv2_2 = nn.Conv2D(
                256, (1,1), strides=stride, padding=(0,0),
                activation=activation, use_bias=use_bias, in_channels=in_channels,
                weight_initializer=mx.init.Xavier(magnitude=2),
                bias_initializer='zeros')
            self.conv2_3 = nn.Conv2D(
                1024, (1,1), strides=stride, padding=(0,0),
                activation=activation, use_bias=use_bias, in_channels=in_channels,
                weight_initializer=mx.init.Xavier(magnitude=2),
                bias_initializer='zeros')
            self.predictor = nn.Conv2D(
                num_channel, kernel, strides=stride, padding=pad,
                activation=activation, use_bias=use_bias, in_channels=in_channels,
                weight_initializer=mx.init.Xavier(magnitude=2),
                bias_initializer='zeros')
            self.bn1 = nn.BatchNorm(epsilon=2e-5)
            self.bn2_1 = nn.BatchNorm(epsilon=2e-5)
            self.bn2_2 = nn.BatchNorm(epsilon=2e-5)
    def hybrid_forward(self, F, x):
        p1_1 = self.bn1(x)
        p1_2 = self.conv1(F.Activation(p1_1, act_type='relu'))
        p2_1 = self.bn2_1(self.conv2_1(F.Activation(p1_1, act_type='relu')))
        p2_2 = self.bn2_2(self.conv2_2(F.Activation(p2_1, act_type='relu')))
        p2_3 = self.conv2_3(F.Activation(p2_2, act_type='relu'))
        return self.predictor(p1_2+p2_3)`
the training information is as follows:
`INFO:root:Namespace(batch_size=8, data_shape=512, dataset='mine', epochs=40, gpus='0', log_interval=100, lr=0.001, lr_decay=0.1, lr_decay_epoch='20,30', momentum=0.9, network='ssd_512_resnet50_v1_voc', num_workers=4, resume='', save_interval=10, save_prefix='myssd3_512_resnet50_v1_voc', seed=233, start_epoch=0, syncbn=False, val_interval=1, wd=0.0005)
INFO:root:Start training from [Epoch 0]
[09:44:36] c:\jenkins\workspace\mxnet\mxnet\src\operator\nn\cudnn\./cudnn_algoreg-inl.h:97: Running performance tests to find the best convolution algorithm, this can take a while... (set the environment variable MXNET_CUDNN_AUTOTUNE_DEFAULT to 0 to disable)
[09:44:57] c:\jenkins\workspace\mxnet\mxnet\src\operator\nn\cudnn\./cudnn_algoreg-inl.h:97: Running performance tests to find the best convolution algorithm, this can take a while... (set the environment variable MXNET_CUDNN_AUTOTUNE_DEFAULT to 0 to disable)
INFO:root:[Epoch 0][Batch 99], Speed: 1.922 samples/sec, CrossEntropy=10147.501, SmoothL1=8.512
INFO:root:[Epoch 0][Batch 199], Speed: 14.907 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 299], Speed: 1.812 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 399], Speed: 1.337 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 499], Speed: 7.995 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 599], Speed: 14.678 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 699], Speed: 1.818 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 799], Speed: 6.643 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 899], Speed: 13.761 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 999], Speed: 2.377 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 1099], Speed: 1.985 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 1199], Speed: 4.003 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 1299], Speed: 1.651 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 1399], Speed: 1.838 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 1499], Speed: 4.293 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 1599], Speed: 6.120 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 1699], Speed: 2.235 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 1799], Speed: 14.980 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 1899], Speed: 2.578 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 1999], Speed: 2.574 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 2099], Speed: 3.524 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 2199], Speed: 15.497 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 2299], Speed: 1.968 samples/sec, CrossEntropy=nan, SmoothL1=nan
INFO:root:[Epoch 0][Batch 2399], Speed: 5.394 samples/sec, CrossEntropy=nan, SmoothL1=nan`

**Comment by Lxh1992** [comment]
Hello! @lsabi @zhreshold Could you help me with this problem? Thank you in advance!
